components/post_edit_history/restore_post_modal.tsx

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo} from 'react';
+import {defineMessages, useIntl} from 'react-intl';
+
+import {t} from 'utils/i18n';
+
+import {Post} from '@mattermost/types/posts';
+import GenericModal from 'components/generic_modal';
+
+const modalMessages = defineMessages({
+    title: {
+        id: t('post_info.edit.restore'),
+        defaultMessage: 'Restore this version',
+    },
+});
+
+type Props = {
+    post: Post;
+    postHeader: JSX.Element;
+    actions: {
+        handleRestore: (post: Post) => void;
+    };
+    onExited: () => void;
+}
+
+const RestorePostModal = ({post, postHeader, actions, onExited}: Props) => {
+    const {formatMessage} = useIntl();
+    const onHide = () => onExited();
+
+    const handleRestore = async () => {
+        await actions.handleRestore(post);
+        onHide();
+    };
+
+    const modalHeaderText = (
+        <div className='edit-post-history__restore__modal__header'>
+            {formatMessage(modalMessages.title) + '?'}
```

Comment:
Why not make the question mark part of the translation string?

Reply:
nice catch

Reply:
I was using the exact same string in other places without question mark. Therefore I tried avoid creating another translation string just for question mark.

Reply:
In that case, I think we probably do want two different versions for those. I wouldn't be surprised if "restore this version" and "restore this version?" are written very differently in some languages

Reply:
That makes sense @hmhealey. Updated as suggested.

Reply:
I asked Tom about this, and he pointed out that, at the very least, this matters in Spanish since they write questions differently since they both start and end with question marks "¿like this?", so that's one place we need separate strings for those